tests/integration/pilot/gw_topology_test.go

```diff
@@ -101,3 +102,175 @@ spec:
 			}
 		})
 }
+
+// If the PROXY protocol is enabled, and no XFF header is provided, client forwarded IPs from the PROXY protocol payload
+// should be used to populate the upstream XFF header.
+//
+// If the PROXY protocol is enabled, and additionally an XFF header is provided AND numTrustedProxies is set, the XFF header on the incoming request
+// will take precedence when populating the upstream XFF header.
+func TestProxyProtocolGatewayEnabled(t *testing.T) {
```

Comment:
Its best to have 1 deployment of the GW and have many subtests. Can we merge this and the one below, or maybe even all 3 into TestXFFGateway?

Deploying a new GW deployment takes ~20s. Deploying a new EnvoyFilter for it takes 100ms. So where possible we should avoid new deployments

Reply:
Considered + initially tried that and felt it resulted in a less obvious set of tests - that creates an implicit ordering for test cases tho (test without filter, apply filter as last test) - rigs don't support "removing" resources except via teardown AFAICT.

Will rework.

Reply:
You can tear down. This is how 99% of our tests work (see TestTraffic, its one test with like 100 sub tests). You just need to use a subtest.


```
Run(
  common setup
  Subtest(
   test something...
  )
  Subtest(
   test something...
  )
  Subtest(
   more setep...
     SubTest(usesetup)
  )
```

Reply:
Done - 1 less gateway created during test run.